src/DotLiquid/StandardFilters.cs

```diff
@@ -166,12 +166,14 @@ public static string H(string input)
         /// <param name="length"></param>
         /// <param name="truncateString"></param>
         /// <returns></returns>
-        public static string Truncate(string input, int length = 50, string truncateString = "...")
+        public static string Truncate(string input, long length = 50, string truncateString = "...")

```

Comment:
Is the right fix to change the type here to long? String Length in .NET is int. There is no secondary 'LongLength' of type long. Memory constrains make that the limit of int max count of character are already not reachable. A string should never reach a length requiring a 64 bit integer to describe.

Reply:
This is really a byproduct of pull request #362. 

The [change](https://github.com/dotliquid/dotliquid/pull/362/files#diff-15f31b73418c0b6c972c740104f051d3R336-R341) to convert any parameters that match a integer regex to a long, means that all functions need to accept long parameters - otherwise we get an ArgumentException even if the value passed in is a valid integer.

I'm not sure if there is any other way to fix this exception other than to revert pull request #362.

Reply:
I reproduced this locally and looking at the call stack I am thinking that Strainer::Invoke or something ahead of it could do safe conversion when the available input argument exceeds the range of required input argument type, with boundary check, so that we don't modify every method that should and can only handle an int32 to handle an int64. I'm wondering if other keywords may have been broken besides Truncate. (And will search for that to answer it too.)

Reply:
This may be a generic approach to handle this for any liquid filter in Strainer::Invoke:

            // Attempt narrowing converstions where required by type mismatch and possible by value range.
            for (var argumentIndex = 0; argumentIndex < parameterInfos.Length; argumentIndex++)
            {
                if (args[argumentIndex] != null && args[argumentIndex].GetType() != parameterInfos[argumentIndex].ParameterType)
                {
                    args[argumentIndex] = Convert.ChangeType(args[argumentIndex], parameterInfos[argumentIndex].ParameterType);
                }
            }

Reply:
We also need to test for IConvertible to avoid trouble with Join filter

            // Attempt conversions where required by type mismatch and possible by value range.
            // These may be narrowing conversions (e.g. Int64 to Int32) when the actual range doesn't cause an overflow.
            for (var argumentIndex = 0; argumentIndex < parameterInfos.Length; argumentIndex++)
            {
                if (args[argumentIndex] != null
                    && args[argumentIndex].GetType() != parameterInfos[argumentIndex].ParameterType
                    && args[argumentIndex] is IConvertible)
                {
                    args[argumentIndex] = Convert.ChangeType(args[argumentIndex], parameterInfos[argumentIndex].ParameterType);
                }
            }

Reply:
Now implemented as recommended